libraries/push/impl/src/main/kotlin/io/element/android/libraries/push/impl/notifications/NotifiableEventResolver.kt

```diff
@@ -238,6 +248,39 @@ class NotifiableEventResolver @Inject constructor(
             stringProvider.getString(R.string.notification_room_invite_body)
         }
     }
+
+    private suspend fun NotificationData.fetchImageIfPresent(client: MatrixClient): Uri? {
+        val fileResult = when (val content = this.content) {
+            is NotificationContent.MessageLike.RoomMessage -> {
+                when (val messageType = content.messageType) {
+                    is AudioMessageType -> null
+                    is VoiceMessageType -> null
+                    is EmoteMessageType -> null
+                    is FileMessageType -> null
+                    is ImageMessageType -> notificationMediaRepoFactory.create(client)
+                        .getMediaFile(
+                            mediaSource = messageType.source,
+                            mimeType = messageType.info?.mimetype,
+                            body = messageType.body,
+                        )
+                    is NoticeMessageType -> null
+                    is TextMessageType -> null
+                    is VideoMessageType -> null
+                    is LocationMessageType -> null
+                    is OtherMessageType -> null
+                }
```

Comment:
Since the method is quite concise (fetch image for image type messages) maybe we could just have a case for `ImageMessageType` and an `else` one?

Reply:
Yes, probably make sense here.

In general I prefer to avoid `else` case in `when` statement to revisit all the `when` statements when there is a new possible case. Also I was wondering if we wanted to use the thumbnail image for `VideoMessageType`, but not sure how to distinguish from images.